apps/rush-lib/src/utilities/TarExecutable.ts

```diff
@@ -28,37 +45,64 @@ export class TarExecutable {
    * @returns
    * The "tar" exit code
    */
-  public async tryUntarAsync(archivePath: string, outputFolderPath: string): Promise<number> {
-    const childProcess: ChildProcess = Executable.spawn(this._tarExecutablePath, ['-x', '-f', archivePath], {
-      currentWorkingDirectory: outputFolderPath
-    });
-    const [tarExitCode] = await events.once(childProcess, 'exit');
-    return tarExitCode;
+  public async tryUntarAsync(options: IUntarOptions): Promise<number> {
+    return await this._spawnTarWithLoggingAsync(
+      ['-x', '-f', options.archivePath],
+      options.outputFolderPath,
+      options.logFilePath
+    );
   }
 
   /**
    * @returns
    * The "tar" exit code
    */
-  public async tryCreateArchiveFromProjectPathsAsync(
-    archivePath: string,
-    paths: string[],
-    project: RushConfigurationProject
-  ): Promise<number> {
+  public async tryCreateArchiveFromProjectPathsAsync(options: ICreateArchiveOptions): Promise<number> {
+    const { project, archivePath, paths, logFilePath } = options;
     const pathsListFilePath: string = `${project.projectRushTempFolder}/tarPaths_${Date.now()}`;
     await FileSystem.writeFileAsync(pathsListFilePath, paths.join('\n'));
 
     const projectFolderPath: string = project.projectFolder;
-    const childProcess: ChildProcess = Executable.spawn(
-      this._tarExecutablePath,
+    const tarExitCode: number = await this._spawnTarWithLoggingAsync(
       ['-c', '-f', archivePath, '-z', '-C', projectFolderPath, '--files-from', pathsListFilePath],
-      {
-        currentWorkingDirectory: projectFolderPath
-      }
+      projectFolderPath,
+      logFilePath
     );
-    const [tarExitCode] = await events.once(childProcess, 'exit');
     await FileSystem.deleteFileAsync(pathsListFilePath);
 
     return tarExitCode;
   }
+
+  private async _spawnTarWithLoggingAsync(
+    args: string[],
+    currentWorkingDirectory: string,
+    logFilePath: string
+  ): Promise<number> {
+    await FileSystem.ensureFolderAsync(path.dirname(logFilePath));
+    const fileWriter: FileWriter = FileWriter.open(logFilePath);
```

Comment:
Can you show an example of what gets written to this file?  (Maybe for both Linux and Windows?)

If it is very large, perhaps we should save the file only if something went wrong.

Reply:
Sure.

Reply:
Tar doesn't actually print anything unless something goes wrong, so this isn't a concern.